distribution/setup.sh

```diff
@@ -0,0 +1,213 @@
+# ------------------------------------------------------------------------------
+# DATADOG_RPM_KEY_CURRENT.public always contains key used to sign current
+# repodata and newly released packages
+# DATADOG_RPM_KEY_FD4BF915.public expires in 2024
+# DATADOG_RPM_KEY_B01082D3.public expires in 2028
+RPM_GPG_KEYS=("DATADOG_RPM_KEY_CURRENT.public" "DATADOG_RPM_KEY_B01082D3.public" "DATADOG_RPM_KEY_FD4BF915.public")
+
+echo -e "\033[34m\n* Vector repository setup script\n\033[0m"
+
+if [ -n "$CSM_MIGRATE" ]; then
+    cloudsmith_migrate=$CSM_MIGRATE
+else
+    cloudsmith_migrate=false
+fi
+
+if [ -n "$VEC_REPO_URL" ]; then
+    repository_url=$VEC_REPO_URL
+else
+    repository_url="vector.dev"
+fi
+
+if [ -n "$TESTING_KEYS_URL" ]; then
+  keys_url=$TESTING_KEYS_URL
+else
+  keys_url="keys.datadoghq.com"
+fi
+
+if [ -n "$TESTING_YUM_URL" ]; then
+  yum_url=$TESTING_YUM_URL
+else
+  yum_url="yum.${repository_url}"
+fi
+
+# We turn off `repo_gpgcheck` for custom REPO_URL, unless explicitly turned
+# on via VEC_RPM_REPO_GPGCHECK.
+# There is more logic for redhat/suse in their specific code branches below
+rpm_repo_gpgcheck=
+if [ -n "$VEC_RPM_REPO_GPGCHECK" ]; then
+    rpm_repo_gpgcheck=$VEC_RPM_REPO_GPGCHECK
+else
+    if [ -n "$REPO_URL" ] || [ -n "$VEC_REPO_URL" ]; then
+        rpm_repo_gpgcheck=0
+    fi
+fi
+
+if [ -n "$TESTING_APT_URL" ]; then
+  apt_url=$TESTING_APT_URL
+else
+  apt_url="apt.${repository_url}"
+fi
+
+vector_major_version=0
+if [ -n "$VECTOR_MAJOR_VERSION" ]; then
+  vector_major_version=$VECTOR_MAJOR_VERSION
+fi
+
+vector_dist_channel=stable
+if [ -n "$VECTOR_DIST_CHANNEL" ]; then
+  vector_dist_channel=$VECTOR_DIST_CHANNEL
+fi
+
+if [ -n "$TESTING_YUM_VERSION_PATH" ]; then
+  yum_version_path=$TESTING_YUM_VERSION_PATH
+else
+  yum_version_path="${vector_dist_channel}/vector-${vector_major_version}"
+fi
+
+if [ -n "$TESTING_APT_REPO_VERSION" ]; then
+  apt_repo_version=$TESTING_APT_REPO_VERSION
+else
+  apt_repo_version="${vector_dist_channel} vector-${vector_major_version}"
+fi
+
+# OS/Distro Detection
+# Try lsb_release, fallback with /etc/issue then uname command
+KNOWN_DISTRIBUTION="(Debian|Ubuntu|RedHat|CentOS|AmazonRocky|AlmaLinux)"
+DISTRIBUTION=$(lsb_release -d 2>/dev/null | grep -Eo $KNOWN_DISTRIBUTION  || grep -Eo $KNOWN_DISTRIBUTION /etc/issue 2>/dev/null || grep -Eo $KNOWN_DISTRIBUTION /etc/Eos-release 2>/dev/null || grep -m1 -Eo $KNOWN_DISTRIBUTION /etc/os-release 2>/dev/null || uname -s)
+
+if [ "$DISTRIBUTION" == "Darwin" ]; then
+    echo -e "\033[31mThis script does not support installing on macOS\033[0m"
+    exit 1;
+
+elif [ -f /etc/debian_version ] || [ "$DISTRIBUTION" == "Debian" ] || [ "$DISTRIBUTION" == "Ubuntu" ]; then
+    OS="Debian"
+elif [ -f /etc/redhat-release ] || [ "$DISTRIBUTION" == "RedHat" ] || [ "$DISTRIBUTION" == "CentOS" ] || [ "$DISTRIBUTION" == "Amazon" ] || [ "$DISTRIBUTION" == "Rocky" ] || [ "$DISTRIBUTION" == "AlmaLinux" ]; then
```

Comment:
It feels strange to have 2 occurences of `Amazon` in 2 different checks, I suppose you could remove the first occurence and keep the 2nd

Reply:
Doesn't it need both because the second `elif` references a different release file?